programs/ssl/dtls_client.c

```diff
@@ -97,6 +97,16 @@ int main(int argc, char *argv[])
     ((void) argc);
     ((void) argv);
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    psa_status_t status = psa_crypto_init();
+    if (status != PSA_SUCCESS) {
+        mbedtls_fprintf(stderr, "Failed to initialize PSA Crypto implementation: %d\n",
+                        (int) status);
+        ret = MBEDTLS_ERR_SSL_HW_ACCEL_FAILED;
+        goto exit;
```

Comment:
This will cause other errors if `psa_crypto_init()` fails: we'll jump to the `exit` label where we'll call a bunch of `mbedtls_xxx_free()` functions on structures on which we haven't called `mbedtls_xxx_init()` yet, so we're going to use uninitialized pointers with the usual kind of unpleasant consequences.

(Same for several other programs.)

This block should be placed after all the `mbedtls_xxx_init()` calls, and the call to `mbedtls_entropy_init()` should be moved together with the other `mbedtls_xxx_init()` calls (as you already did in one place).